istioctl/cmd/analyze.go

```diff
@@ -257,12 +265,16 @@ istioctl experimental analyze -k -d false
 			"Analyzers requiring resources made available by enabling service discovery will be skipped.")
 	analysisCmd.PersistentFlags().BoolVar(&colorize, "color", istioctlColorDefault(analysisCmd),
 		"Default true.  Disable with '=false' or set $TERM to dumb")
-	analysisCmd.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Enable verbose output")
+	analysisCmd.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false,
+		"Enable verbose output")
 	analysisCmd.PersistentFlags().Var(&failureLevel, "failure-threshold",
 		fmt.Sprintf("The severity level of analysis at which to set a non-zero exit code. Valid values: %v", diag.GetAllLevelStrings()))
 	analysisCmd.PersistentFlags().Var(&outputLevel, "output-threshold",
 		fmt.Sprintf("The severity level of analysis at which to display messages. Valid values: %v", diag.GetAllLevelStrings()))
-	analysisCmd.PersistentFlags().StringVarP(&msgOutputFormat, "output", "o", LogOutput, fmt.Sprintf("Output format: one of %v", msgOutputFormatKeys))
+	analysisCmd.PersistentFlags().StringVarP(&msgOutputFormat, "output", "o", LogOutput,
+		fmt.Sprintf("Output format: one of %v", msgOutputFormatKeys))
+	analysisCmd.PersistentFlags().StringVar(&meshCfgFile, "mesh-config-file", "",
+		"Overrides the mesh config values to use for analysis.")
```

Comment:
`istioctl` isn't consistent in its usage of lowercase-dash or camelCase for args. (e.g. it's `kube-inject` instead of `kubeInject`. From what I've seen `istioctl` uses the dash format more commonly than camelCase, and `istioctl analyze` certainly does. So does `kubectl` and the vast majority of CLI tools out there. 

Given all that, I think I'd prefer to keep it as it is, but if you feel strongly I can change it.

Reply:
Agreed that istioctl is generally inconsistent about camelCase vs. lowercase-dash. We should try to fix that. Something like https://godoc.org/github.com/spf13/pflag#FlagSet.SetNormalizeFunc may help ease the transition. cc @esnible @therealmitchconnors 

However, in this case camelCase is consistently used for the mesh config filename. IMO consistency at the per-flag level is preferable to partial consistency at the global level. 

```bash
$ grep -rn -e '"meshConfigFile"' istioctl/
istioctl/cmd/add-to-mesh.go:125:        cmd.PersistentFlags().StringVar(&meshConfigFile, "meshConfigFile", "",
istioctl/cmd/kubeinject.go:318: injectCmd.PersistentFlags().StringVar(&meshConfigFile, "meshConfigFile", "",
istioctl/cmd/authz.go:312:      convertCmd.PersistentFlags().StringVarP(&meshConfig, "meshConfigFile", "m", "",
$ grep -rn -e '"mesh-config-file"' istioctl/ | wc -l
0
```

Reply:
That makes sense to me. Changed it.